client/components/pack/AddPack.tsx

```diff
@@ -3,34 +3,34 @@ import { Box, Input, Button, Text, Select, CheckIcon } from 'native-base';
 
 // import useAddPack from "../../hooks/useAddPack";
 import { addPack } from '../../store/packsStore';
-import { useState } from 'react';
+import {  useState } from 'react';
 // import { useAuth } from "../../auth/provider";
 import { useSelector, useDispatch } from 'react-redux';
 import { CustomModal } from '../modal';
 import useTheme from '../../hooks/useTheme';
 import useCustomStyles from '~/hooks/useCustomStyles';
 import { useAddNewPack } from '~/hooks/packs';
+import { useRouter } from 'expo-router';
 
-export const AddPack = () => {
+export const AddPack = ({isCreatingTrip}) => {
   const { enableDarkMode, enableLightMode, isDark, isLight, currentTheme } =
     useTheme();
   const styles = useCustomStyles(loadStyles);
 
   const dispatch = useDispatch();
+  const router = useRouter();
 
   const [name, setName] = useState('');
   const [isPublic, setIsPublic] = useState(false);
 
   // const { addPack } = useAddPack();
   // const { user } = useAuth();
   const user = useSelector((state) => state.auth.user);
-
   const isLoading = useSelector((state) => state.packs.isLoading);
-
-  const error = useSelector((state) => state.packs.error);
-
-  const isError = error !== null;
-  const { addNewPack } = useAddNewPack();
+  const { addNewPack,isSuccess,isError } = useAddNewPack();
+    if (isSuccess && !isCreatingTrip ) {
+      router.push('/packs');
```

Comment:
we should get the id from response and direct to there. double check but i think it will be singular /pack/:packId for the route